<a href="https://colab.research.google.com/github/MFaiqKhan/classical_MachineLearning/blob/main/KFold_IrisDataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [2]:
iris = load_iris()

**Logistic Regression**

In [3]:
l_scores = cross_val_score(LogisticRegression(), iris.data, iris.target)
l_scores

/usr/local/lib/python3.9/dist-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [4]:
np.average(l_scores)

0.9733333333333334

**Decision Tree**

In [5]:
tree = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
tree

array([0.96666667, 0.96666667, 0.9       , 0.93333333, 1.        ])

In [6]:
np.average(tree)

0.9533333333333334

**Support Vector Machines (SVM)**

In [8]:
SVC_scores = cross_val_score(SVC(), iris.data, iris.target)
SVC_scores

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

In [9]:
np.average(SVC_scores)

0.9666666666666666

**Random Forest**

In [10]:
rf_scores = cross_val_score(RandomForestClassifier(), iris.data, iris.target)
rf_scores

array([0.96666667, 0.96666667, 0.93333333, 0.93333333, 1.        ])

In [11]:
np.average(rf_scores)

0.96

**Best Performing Model is of Logistic Regression with Score : 0.9733333333333334**